### ISSUE TITLE
Normalizing query using reference SCT model Error in umi[residual.features, , drop = FALSE] :    invalid or not-yet-implemented 'Matrix' subsetting

### ISSUE THREAD

**Issue by Elham-adabi** [satijalab/seurat#6063]
Dear Seurat team,
I have transcriptome data together with surface protein data (ABSEQ) of PBMCs and I run the multimodel data analysis and WNN, I could also run reference mapping before, but suddenly it stopped working and I get the following error :
```{r}
anchors <- FindTransferAnchors(
  reference = reference,
  query = cart.combined,
  normalization.method = "SCT",
  reference.reduction = "spca",
  dims = 1:50
)
```
Normalizing query using reference SCT model
Error in umi[residual.features, , drop = FALSE] : 
  invalid or not-yet-implemented 'Matrix' subsetting

Here are my session info and more details of my coding: 
─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.2.0 (2022-04-22 ucrt)
 os       Windows 10 x64 (build 22000)
 system   x86_64, mingw32
 ui       RStudio
 language (EN)
 collate  English_United States.utf8
 ctype    English_United States.utf8
 tz       Europe/Berlin
 date     2022-06-10
 rstudio  2022.02.3+492 Prairie Trillium (desktop)
 pandoc   2.17.1.1 @ C:/Program Files/RStudio/bin/quarto/bin/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 package          * version    date (UTC) lib source
 abind              1.4-5      2016-07-21 [1] CRAN (R 4.2.0)
 assertthat         0.2.1      2019-03-21 [1] CRAN (R 4.2.0)
 backports          1.4.1      2021-12-13 [1] CRAN (R 4.2.0)
 BiocGenerics       0.42.0     2022-04-26 [1] Bioconductor
 BiocManager        1.30.18    2022-05-18 [1] CRAN (R 4.2.0)
 BiocParallel       1.30.3     2022-06-07 [1] Bioconductor
 Biostrings         2.64.0     2022-04-26 [1] Bioconductor
 bit                4.0.4      2020-08-04 [1] CRAN (R 4.2.0)
 bit64              4.0.5      2020-08-30 [1] CRAN (R 4.2.0)
 bitops             1.0-7      2021-04-24 [1] CRAN (R 4.2.0)
 brio               1.1.3      2021-11-30 [1] CRAN (R 4.2.0)
 broom              0.8.0      2022-04-13 [1] CRAN (R 4.2.0)
 cachem             1.0.6      2021-08-19 [1] CRAN (R 4.2.0)
 callr              3.7.0      2021-04-20 [1] CRAN (R 4.2.0)
 cellranger         1.1.0      2016-07-27 [1] CRAN (R 4.2.0)
 cli                3.3.0      2022-04-25 [1] CRAN (R 4.2.0)
 cluster            2.1.3      2022-03-28 [2] CRAN (R 4.2.0)
 codetools          0.2-18     2020-11-04 [2] CRAN (R 4.2.0)
 colorspace         2.0-3      2022-02-21 [1] CRAN (R 4.2.0)
 cowplot            1.1.1      2020-12-30 [1] CRAN (R 4.2.0)
 crayon             1.5.1      2022-03-26 [1] CRAN (R 4.2.0)
 curl               4.3.2      2021-06-23 [1] CRAN (R 4.2.0)
 data.table         1.14.2     2021-09-27 [1] CRAN (R 4.2.0)
 DBI                1.1.2      2021-12-20 [1] CRAN (R 4.2.0)
 dbplyr             2.2.0      2022-06-05 [1] CRAN (R 4.2.0)
 deldir             1.0-6      2021-10-23 [1] CRAN (R 4.2.0)
 desc               1.4.1      2022-03-06 [1] CRAN (R 4.2.0)
 devtools           2.4.3      2021-11-30 [1] CRAN (R 4.2.0)
 digest             0.6.29     2021-12-01 [1] CRAN (R 4.2.0)
 diptest            0.76-0     2021-05-04 [1] CRAN (R 4.2.0)
 dplyr            * 1.0.9      2022-04-28 [1] CRAN (R 4.2.0)
 ellipsis           0.3.2      2021-04-29 [1] CRAN (R 4.2.0)
 evaluate           0.15       2022-02-18 [1] CRAN (R 4.2.0)
 fansi              1.0.3      2022-03-24 [1] CRAN (R 4.2.0)
 farver             2.1.0      2021-02-28 [1] CRAN (R 4.2.0)
 fastmap            1.1.0      2021-01-25 [1] CRAN (R 4.2.0)
 fastmatch          1.1-3      2021-07-23 [1] CRAN (R 4.2.0)
 fitdistrplus       1.1-8      2022-03-10 [1] CRAN (R 4.2.0)
 forcats          * 0.5.1      2021-01-27 [1] CRAN (R 4.2.0)
 fs                 1.5.2      2021-12-08 [1] CRAN (R 4.2.0)
 future             1.26.1     2022-05-27 [1] CRAN (R 4.2.0)
 future.apply       1.9.0      2022-04-25 [1] CRAN (R 4.2.0)
 generics           0.1.2      2022-01-31 [1] CRAN (R 4.2.0)
 GenomeInfoDb       1.32.2     2022-05-15 [1] Bioconductor
 GenomeInfoDbData   1.2.8      2022-06-02 [1] Bioconductor
 GenomicRanges      1.48.0     2022-04-26 [1] Bioconductor
 ggplot2          * 3.3.6      2022-05-03 [1] CRAN (R 4.2.0)
 ggrepel          * 0.9.1      2021-01-15 [1] CRAN (R 4.2.0)
 ggridges           0.5.3      2021-01-08 [1] CRAN (R 4.2.0)
 globals            0.15.0     2022-05-09 [1] CRAN (R 4.2.0)
 glue               1.6.2      2022-02-24 [1] CRAN (R 4.2.0)
 goftest            1.2-3      2021-10-07 [1] CRAN (R 4.2.0)
 gridExtra          2.3        2017-09-09 [1] CRAN (R 4.2.0)
 gtable             0.3.0      2019-03-25 [1] CRAN (R 4.2.0)
 haven              2.5.0      2022-04-15 [1] CRAN (R 4.2.0)
 hdf5r            * 1.3.5      2022-06-02 [1] Github (hhoeflin/hdf5r@fc62b28)
 hms                1.1.1      2021-09-26 [1] CRAN (R 4.2.0)
 htmltools          0.5.2      2021-08-25 [1] CRAN (R 4.2.0)
 htmlwidgets        1.5.4      2021-09-08 [1] CRAN (R 4.2.0)
 httpuv             1.6.5      2022-01-05 [1] CRAN (R 4.2.0)
 httr               1.4.3      2022-05-04 [1] CRAN (R 4.2.0)
 ica                1.0-2      2018-05-24 [1] CRAN (R 4.2.0)
 igraph             1.3.1      2022-04-20 [1] CRAN (R 4.2.0)
 IRanges            2.30.0     2022-04-26 [1] Bioconductor
 irlba              2.3.5      2021-12-06 [1] CRAN (R 4.2.0)
 jsonlite           1.8.0      2022-02-22 [1] CRAN (R 4.2.0)
 KernSmooth         2.23-20    2021-05-03 [2] CRAN (R 4.2.0)
 knitr              1.39       2022-04-26 [1] CRAN (R 4.2.0)
 ks                 1.13.5     2022-04-14 [1] CRAN (R 4.2.0)
 labeling           0.4.2      2020-10-20 [1] CRAN (R 4.2.0)
 later              1.3.0      2021-08-18 [1] CRAN (R 4.2.0)
 lattice            0.20-45    2021-09-22 [2] CRAN (R 4.2.0)
 lazyeval           0.2.2      2019-03-15 [1] CRAN (R 4.2.0)
 leiden             0.4.2      2022-05-09 [1] CRAN (R 4.2.0)
 lifecycle          1.0.1      2021-09-24 [1] CRAN (R 4.2.0)
 listenv            0.8.0      2019-12-05 [1] CRAN (R 4.2.0)
 lmtest             0.9-40     2022-03-21 [1] CRAN (R 4.2.0)
 lubridate          1.8.0      2021-10-07 [1] CRAN (R 4.2.0)
 magrittr           2.0.3      2022-03-30 [1] CRAN (R 4.2.0)
 MASS               7.3-56     2022-03-23 [2] CRAN (R 4.2.0)
 Matrix           * 1.4-1      2022-03-23 [1] CRAN (R 4.2.0)
 matrixStats        0.62.0     2022-04-19 [1] CRAN (R 4.2.0)
 mclust             5.4.10     2022-05-20 [1] CRAN (R 4.2.0)
 memoise            2.0.1      2021-11-26 [1] CRAN (R 4.2.0)
 mgcv               1.8-40     2022-03-29 [2] CRAN (R 4.2.0)
 mime               0.12       2021-09-28 [1] CRAN (R 4.2.0)
 miniUI             0.1.1.1    2018-05-18 [1] CRAN (R 4.2.0)
 modelr             0.1.8      2020-05-19 [1] CRAN (R 4.2.0)
 multimode        * 1.5        2021-03-20 [1] CRAN (R 4.2.0)
 munsell            0.5.0      2018-06-12 [1] CRAN (R 4.2.0)
 mvtnorm            1.1-3      2021-10-08 [1] CRAN (R 4.2.0)
 nlme               3.1-157    2022-03-25 [2] CRAN (R 4.2.0)
 parallelly         1.32.0     2022-06-07 [1] CRAN (R 4.2.0)
 patchwork        * 1.1.1      2020-12-17 [1] CRAN (R 4.2.0)
 pbapply            1.5-0      2021-09-16 [1] CRAN (R 4.2.0)
 pillar             1.7.0      2022-02-01 [1] CRAN (R 4.2.0)
 pkgbuild           1.3.1      2021-12-20 [1] CRAN (R 4.2.0)
 pkgconfig          2.0.3      2019-09-22 [1] CRAN (R 4.2.0)
 pkgload            1.2.4      2021-11-30 [1] CRAN (R 4.2.0)
 plotly             4.10.0     2021-10-09 [1] CRAN (R 4.2.0)
 plyr               1.8.7      2022-03-24 [1] CRAN (R 4.2.0)
 png                0.1-7      2013-12-03 [1] CRAN (R 4.2.0)
 polyclip           1.10-0     2019-03-14 [1] CRAN (R 4.2.0)
 pracma             2.3.8      2022-03-04 [1] CRAN (R 4.2.0)
 prettyunits        1.1.1      2020-01-24 [1] CRAN (R 4.2.0)
 processx           3.5.3      2022-03-25 [1] CRAN (R 4.2.0)
 progressr          0.10.1     2022-06-03 [1] CRAN (R 4.2.0)
 promises           1.2.0.1    2021-02-11 [1] CRAN (R 4.2.0)
 ps                 1.7.0      2022-04-23 [1] CRAN (R 4.2.0)
 purrr            * 0.3.4      2020-04-17 [1] CRAN (R 4.2.0)
 R6                 2.5.1      2021-08-19 [1] CRAN (R 4.2.0)
 RANN               2.6.1      2019-01-08 [1] CRAN (R 4.2.0)
 RColorBrewer       1.1-3      2022-04-03 [1] CRAN (R 4.2.0)
 Rcpp               1.0.8.3    2022-03-17 [1] CRAN (R 4.2.0)
 RcppAnnoy          0.0.19     2021-07-30 [1] CRAN (R 4.2.0)
 RcppRoll           0.3.0      2018-06-05 [1] CRAN (R 4.2.0)
 RCurl              1.98-1.7   2022-06-09 [1] CRAN (R 4.2.0)
 readr            * 2.1.2      2022-01-30 [1] CRAN (R 4.2.0)
 readxl             1.4.0      2022-03-28 [1] CRAN (R 4.2.0)
 remotes            2.4.2      2021-11-30 [1] CRAN (R 4.2.0)
 reprex             2.0.1      2021-08-05 [1] CRAN (R 4.2.0)
 reshape2           1.4.4      2020-04-09 [1] CRAN (R 4.2.0)
 reticulate         1.25       2022-05-11 [1] CRAN (R 4.2.0)
 rgeos              0.5-9      2021-12-15 [1] CRAN (R 4.2.0)
 rlang              1.0.2      2022-03-04 [1] CRAN (R 4.2.0)
 rmarkdown          2.14       2022-04-25 [1] CRAN (R 4.2.0)
 ROCR               1.0-11     2020-05-02 [1] CRAN (R 4.2.0)
 rootSolve          1.8.2.3    2021-09-29 [1] CRAN (R 4.2.0)
 rpart              4.1.16     2022-01-24 [2] CRAN (R 4.2.0)
 rprojroot          2.0.3      2022-04-02 [1] CRAN (R 4.2.0)
 Rsamtools          2.12.0     2022-04-26 [1] Bioconductor
 RSpectra           0.16-1     2022-04-24 [1] CRAN (R 4.2.0)
 rstudioapi         0.13       2020-11-12 [1] CRAN (R 4.2.0)
 Rtsne              0.16       2022-04-17 [1] CRAN (R 4.2.0)
 rvest              1.0.2      2021-10-16 [1] CRAN (R 4.2.0)
 S4Vectors          0.34.0     2022-04-26 [1] Bioconductor
 scales             1.2.0      2022-04-13 [1] CRAN (R 4.2.0)
 scattermore        0.8        2022-02-14 [1] CRAN (R 4.2.0)
 sctransform      * 0.3.3      2022-06-10 [1] Github (satijalab/sctransform@e9e52a4)
 sessioninfo        1.2.2      2021-12-06 [1] CRAN (R 4.2.0)
 Seurat           * 4.1.1      2022-05-02 [1] CRAN (R 4.2.0)
 SeuratDisk       * 0.0.0.9020 2022-06-10 [1] Github (mojaveazure/seurat-disk@9b89970)
 SeuratObject     * 4.1.0      2022-05-01 [1] CRAN (R 4.2.0)
 shiny              1.7.1      2021-10-02 [1] CRAN (R 4.2.0)
 Signac           * 1.7.0      2022-06-01 [1] CRAN (R 4.2.0)
 sp               * 1.5-0      2022-06-05 [1] CRAN (R 4.2.0)
 spatstat.core      2.4-4      2022-05-18 [1] CRAN (R 4.2.0)
 spatstat.data      2.2-0      2022-04-18 [1] CRAN (R 4.2.0)
 spatstat.geom      2.4-0      2022-03-29 [1] CRAN (R 4.2.0)
 spatstat.random    2.2-0      2022-03-30 [1] CRAN (R 4.2.0)
 spatstat.sparse    2.1-1      2022-04-18 [1] CRAN (R 4.2.0)
 spatstat.utils     2.3-1      2022-05-06 [1] CRAN (R 4.2.0)
 stringi            1.7.6      2021-11-29 [1] CRAN (R 4.2.0)
 stringr          * 1.4.0      2019-02-10 [1] CRAN (R 4.2.0)
 survival           3.3-1      2022-03-03 [2] CRAN (R 4.2.0)
 tensor             1.5        2012-05-05 [1] CRAN (R 4.2.0)
 testthat           3.1.4      2022-04-26 [1] CRAN (R 4.2.0)
 tibble           * 3.1.7      2022-05-03 [1] CRAN (R 4.2.0)
 tidyr            * 1.2.0      2022-02-01 [1] CRAN (R 4.2.0)
 tidyselect         1.1.2      2022-02-21 [1] CRAN (R 4.2.0)
 tidyverse        * 1.3.1      2021-04-15 [1] CRAN (R 4.2.0)
 tzdb               0.3.0      2022-03-28 [1] CRAN (R 4.2.0)
 usethis            2.1.6      2022-05-25 [1] CRAN (R 4.2.0)
 utf8               1.2.2      2021-07-24 [1] CRAN (R 4.2.0)
 uwot               0.1.11     2021-12-02 [1] CRAN (R 4.2.0)
 vctrs              0.4.1      2022-04-13 [1] CRAN (R 4.2.0)
 viridisLite        0.4.0      2021-04-13 [1] CRAN (R 4.2.0)
 vroom              1.5.7      2021-11-30 [1] CRAN (R 4.2.0)
 withr              2.5.0      2022-03-03 [1] CRAN (R 4.2.0)
 xfun               0.31       2022-05-10 [1] CRAN (R 4.2.0)
 xml2               1.3.3      2021-11-30 [1] CRAN (R 4.2.0)
 xtable             1.8-4      2019-04-21 [1] CRAN (R 4.2.0)
 XVector            0.36.0     2022-04-26 [1] Bioconductor
 yaml               2.3.5      2022-02-21 [1] CRAN (R 4.2.0)
 zlibbioc           1.42.0     2022-04-26 [1] Bioconductor
 zoo                1.8-10     2022-04-15 [1] CRAN (R 4.2.0)

 cart_combined.list <- lapply(X = cart_combined.list, FUN = function(x) {
+     x <- NormalizeData(x)
+     x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 423, Vars.to.regress = "nCount_RNA")
+ })
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
Warning: The following arguments are not used: Vars.to.regress
Calculating gene variances
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
Calculating feature variances of standardized and clipped values
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
Warning: The following arguments are not used: Vars.to.regress
Calculating gene variances
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
Calculating feature variances of standardized and clipped values
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
> features <- SelectIntegrationFeatures(object.list = cart_combined.list)
> cart.anchors <- FindIntegrationAnchors(object.list = cart_combined.list, anchor.features = features)
Scaling features for provided objects
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100% elapsed=02s  
Finding all pairwise anchors
  |                                                  | 0 % ~calculating  Running CCA
Merging objects
Finding neighborhoods
Finding anchors
	Found 49142 anchors
Filtering anchors
	Retained 14171 anchors
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100% elapsed=07m 09s
> cart.combined <- IntegrateData(anchorset = cart.anchors)
Merging dataset 1 into 2
Extracting anchors for merged samples
Finding integration vectors
Finding integration vector weights
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
> cart.combined <- FindNeighbors(cart.combined, reduction = "pca", dims = 1:15)
Computing nearest neighbor graph
Computing SNN
> cart.combined <- FindClusters(cart.combined, resolution = 0.5)
> cart.combined <- NormalizeData(cart.combined, normalization.method = "CLR", assay = "ABSEQ")
Normalizing across features
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100% elapsed=00s 
> cart.combined <- FindVariableFeatures(cart.combined, selection.method = "vst",assay = "ABSEQ", nfeatures = 33, vars.to.regress = "nCount_ABSEQ")
Warning: The following arguments are not used: vars.to.regress
Calculating gene variances
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
Calculating feature variances of standardized and clipped values
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
> cart.combined <- ScaleData(cart.combined, assay = "ABSEQ")
Centering and scaling data matrix
  |======================================================================================================================================| 100%
> cart.combined <- RunPCA(cart.combined, npcs = 50, verbose = FALSE, reduction.name = 'apca', assay = "ABSEQ", approx = FALSE )
Warning: Cannot add objects with duplicate keys (offending key: PC_), setting key to 'apca_'
> cart.combined <- FindMultiModalNeighbors(
+   cart.combined, reduction.list = list("pca", "apca"), 
+   dims.list = list(1:15, 1:15), modality.weight.name = "RNA.weight"
+ )
Calculating cell-specific modality weights
Finding 20 nearest neighbors for each modality.
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100% elapsed=16s  
Calculating kernel bandwidths
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100% elapsed=02s  
Warning in FindMultiModalNeighbors(cart.combined, reduction.list = list("pca",  :
  The number of provided modality.weight.name is not equal to the number of modalities. integrated.weight ABSEQ.weight are used to store the modality weights
Finding multimodal neighbors
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100% elapsed=01m 07s
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100% elapsed=02s  
Constructing multimodal KNN graph
Constructing multimodal SNN graph
> cart.combined <- RunUMAP(cart.combined, nn.name = "weighted.nn", reduction.name = "wnn.umap", reduction.key = "wnnUMAP_")
16:46:02 UMAP embedding parameters a = 0.9922 b = 1.112
16:46:03 Commencing smooth kNN distance calibration using 1 thread
16:46:05 Initializing from normalized Laplacian + noise
16:46:06 Commencing optimization for 200 epochs, with 1097058 positive edges
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
16:46:38 Optimization finished
> cart.combined <- FindClusters(cart.combined, graph.name = "wsnn", algorithm = 3, resolution = 2, verbose = FALSE)
> 
> 
> reference <- LoadH5Seurat("C:/Sequencing data/R-analysis/CarT/pbmc_multimodal.h5seurat")
Validating h5Seurat file
Initializing ADT with data
Adding counts for ADT
Adding variable feature information for ADT
Adding miscellaneous information for ADT
Initializing SCT with data
Adding counts for SCT
Adding variable feature information for SCT
Adding miscellaneous information for SCT
Adding reduction apca
Adding cell embeddings for apca
Adding feature loadings for apca
Adding miscellaneous information for apca
Adding reduction aumap
Adding cell embeddings for aumap
Adding miscellaneous information for aumap
Adding reduction pca
Adding cell embeddings for pca
Adding feature loadings for pca
Adding miscellaneous information for pca
Adding reduction spca
Adding cell embeddings for spca
Adding feature loadings for spca
Adding miscellaneous information for spca
Adding reduction umap
Adding cell embeddings for umap
Adding miscellaneous information for umap
Adding reduction wnn.umap
Adding cell embeddings for wnn.umap
Adding miscellaneous information for wnn.umap
Adding graph wknn
Adding graph wsnn
Adding command information
Adding cell-level metadata
Adding miscellaneous information
Adding tool-specific results
> anchors <- FindTransferAnchors(
+   reference = reference,
+   query = cart.combined,
+   normalization.method = "SCT",
+   reference.reduction = "spca",
+   dims = 1:50
+ )
Normalizing query using reference SCT model
Error in umi[residual.features, , drop = FALSE] : 
  invalid or not-yet-implemented 'Matrix' subsetting
 
I would appreciate your help.

**Comment by saketkc** [comment]
What is the output of `class(GetAssayData(object = cart.combined, assay="RNA", slot = "counts"))`

**Comment by Elham-adabi** [comment]
Dear Saket, 
Thanks for your reply, here is the output:
> class(GetAssayData(object = cart.combined, assay="RNA", slot = "counts"))
[1] "dgCMatrix"
attr(,"package")
[1] "Matrix"

**Comment by saketkc** [comment]
Thanks. I am unable to reproduce this at my end. But if you can share an object, that would be helpful. My email is schoudhary@nygenome.org

**Comment by Elham-adabi** [comment]
Thanks a lot for looking into that, 
I can send you an object but I suspect that it is coming from a package that I have/ I don't have cause this was working for me before with the same data set and script and stopped working with the mentioned error after adding new packages and updating my old packages. Do you have any idea about this?

I also came up with another question regarding the normalization. I normalize my RNA data with Seurat log normalization and the Antibody data with CLR, as it is in the https://satijalab.org/seurat/articles/multimodal_vignette.html
but during the mapping to the multimodal reference, normalization. method = "SCT" is used. I am wondering how this different normalization method may affect the mapping results?

**Comment by tkim-NW** [comment]
@Elham-adabi for your second question if you are using log normalization to normalize RNA data and not SCTranform, then you shouldn't specify normalization. method = "SCT" when mapping to the multimodal reference, this option is if you did do SCTranform normalization. You'll probably want to use normalization.method = "LogNormalize"

**Comment by Elham-adabi** [comment]
@tkim-w thanks for your reply, I tried it but then get the following error, I wonder if there is any LogNormalize reference for this cases?

Error: An SCT assay (SCT) was provided for reference.assay but normalization.method was set as LogNormalize.

**Comment by tkim-w** [comment]
@Elham-adabi 
Oop that's my bad, it seems that the normalization you use on your query data should match the normalization used by the reference. Check out #3777 and [this comment](https://github.com/satijalab/seurat/issues/3777#issuecomment-739031368).

**Comment by saketkc** [comment]
@Elham-adabi thanks for sharing the data. Your `ABSEQ` (also your default assay) assay has no common features with the reference:
```r
length(intersect(rownames(cart.combined), rownames(reference)))
>0
```

You will just need to specify the assay: `FindTransferAnchors(..., query.assay="RNA")`.

**Comment by Elham-adabi** [comment]
@saketkc Thanks a lot, It is working now. 
I was wondering if that means my ABSEQ data will not be used to annotate clusters or when transferring cell type labels and protein data from the reference to the query they will be used?
cart.combined <- MapQuery(
  anchorset = anchors,
  query = cart.combined,
  reference = reference,
  refdata = list(
    celltype.l1 = "celltype.l1",
    celltype.l2 = "celltype.l2",
    celltype.l3 = "celltype.l3",
    predicted_ADT = "ABSEQ"
  ),
  reference.reduction = "spca", 
  reduction.model = "wnn.umap"
)

**Comment by saketkc** [comment]
That's correct. The reference umap is based on both modalities, but when you transfer only the RNA modality is used such that the separation between celltypes (based on both modalities) is maximized (that is what spca tries to achieve).

**Comment by Angel-Wei** [comment]
I had the same error when I did label transfer and found threads on this post. After trouble shooting, in my case, the reason why I got this error is because the reference and the query objects are using different feature identifiers: the reference obj is using Ensembl gene id while my query object is using gene name. Just giving some information here, if anybody might have similar situation like I do and got the same problem!
